data/bot_api.lua

```diff
@@ -158,12 +158,11 @@ end
 function blob_time_to_y (destination) 
 	-- TODO allow specifying whether upward or downward!
 	-- TODO error handling
-	local blobbygroundpos = CONST_GROUND_HEIGHT + CONST_BLOBBY_HEIGHT / 2 
-	local y = posy()
-	grav = CONST_BLOBBY_GRAVITY / 2    -- half, because we use jump buffer
-	time1 = CONST_BLOBBY_JUMP/grav + math.sqrt(2*grav*(y-blobbygroundpos) + CONST_BLOBBY_JUMP*CONST_BLOBBY_JUMP) / grav
-	time2 = CONST_BLOBBY_JUMP/grav - math.sqrt(2*grav*(y-blobbygroundpos) + CONST_BLOBBY_JUMP*CONST_BLOBBY_JUMP) / grav
-	timemin = math.min(time1,time2)
+	local blobbygroundpos = CONST_GROUND_HEIGHT + CONST_BLOBBY_HEIGHT / 2
+	local grav = CONST_BLOBBY_GRAVITY / 2    -- half, because we use jump buffer
+	local time1 = CONST_BLOBBY_JUMP/grav + math.sqrt(2*grav*(destination-blobbygroundpos) + CONST_BLOBBY_JUMP*CONST_BLOBBY_JUMP) / grav
+	local time2 = CONST_BLOBBY_JUMP/grav - math.sqrt(2*grav*(destination-blobbygroundpos) + CONST_BLOBBY_JUMP*CONST_BLOBBY_JUMP) / grav
+	local timemin = math.min(time1, time2)
 	return timemin
 end
```

Comment:
good question :)
I guess both could be valuable. 

From what I can see, it is only used in 
```
	if blob_time_to_y(by) <= time + 1 or launched() then
		jump()
	end
```
where it actually does not matter because of the `launched()` call. But maybe using current-position to destination would be the better option, and consistent with `blob_time_to_x`.

Reply:
I think [2c2240a](https://github.com/danielknobe/blobbyvolley2/pull/101/commits/2c2240a00419e1b72645e4826b31a230d4136a7b) is not going to work. I have written down the formula:
https://github.com/danielknobe/blobbyvolley2/blob/doc/physics/doc/physics/constant%20acceleration%20motion.pdf
If we want to calculate from any position we need the current velocity instead of start velocity.

Reply:
you are right. The problem is even trickier, though. It seems that we give `CONST_BLOBBY_JUMP` the wrong sign when creating this constant in lua (i.e., for all quantities, we take care to invert the y-axis when converting from c++ to lua, but here we do not). 
In addition, yes, we should take the current velocity, unless we are standing on the ground, then we need the jump velocity.